### PR TITLE
Mempool txn prioritization

### PR DESCRIPTION
**Pull request by viquezclaudio**
- Introduces some priorities (low, medium, high) to transactions when they are added into the mempool.
- Exposes this functionality via RPC
- The unpark transaction is included into the mempool with high priority

### CODE REVIEW

File: mempool/src/mempool.rs [hunk 1]
```diff
@@ -668,7 +673,15 @@ impl Mempool {
...
             Ok(mempool_state_lock) => {
-                RwLockUpgradableReadGuard::upgrade(mempool_state_lock).put(&transaction);
+                if let Some(priority) = tx_priority {
+                    RwLockUpgradableReadGuard::upgrade(mempool_state_lock)
+                        .put(&transaction, priority);
+                } else {
+                    //If the priority is not provided we default to medium
+                    RwLockUpgradableReadGuard::upgrade(mempool_state_lock)
+                        .put(&transaction, TxPriority::MediumPriority);
+                }
```
Comment (nibhar) [inline mempool/src/mempool.rs]:
```suggestion
                RwLockUpgradableReadGuard::upgrade(mempool_state_lock)
                        .put(&transaction, tx_priority.unwrap_or(TxPriority::MediumPriority));
```

File: mempool/src/mempool.rs [hunk 2]
```diff
@@ -520,7 +521,7 @@ impl Mempool {
                     let pending_balance = tx.total_value() + sender_total;
 
                     if pending_balance <= sender_balance {
-                        mempool_state.put(tx);
+                        mempool_state.put(tx, TxPriority::MediumPriority);
```
Comment (nibhar) [inline mempool/src/mempool.rs:525]: All reverted transactions are added as `MediumPriority`. The fact they might have been added as `HighPriority` is lost of course, but potentially we could still add unpark transactions as `HighPriority`.

**Review by nibhar — APPROVED** [review]
Good to go! My comments just to save some code and a question for trannsactions of reverted blocks.

**Comment by sisou** [comment]
When querying mempool state, e.g. via RPC `mempoolContent()` method, are the transactions ordered by priority? Is the priority part of the returned object (if I set `include_transactions` to `true`)?

**Pull request merged**